src/Microsoft.Diagnostics.Runtime/src/Common/HeapBase.cs

```diff
@@ -257,6 +259,34 @@ protected internal override IEnumerable<ClrObject> EnumerateObjectReferences(ulo
             return result;
         }
 
+        protected internal override IEnumerable<ClrObjectReference> EnumerateObjectReferencesWithFields(ulong obj, ClrType type, bool carefully)
+        {
+            if (type == null)
+                type = GetObjectType(obj);
+            else
+                Debug.Assert(type == GetObjectType(obj));
+
+            if (!type.ContainsPointers)
```

Comment:
`GetObjectType` can return null if the heap is in an inconsistent state.  This if check should be `if (type == null || !type.ContainsPointers)`.